operator/pkg/resources/statefulset_update.go

```diff
@@ -53,6 +56,21 @@ var (
 	errUnderReplicatedPartition = errors.New("partition under replicated")
 )
 
+func (r *StatefulSetResource) getNodePoolStatus() vectorizedv1alpha1.NodePoolStatus {
+	if r.pandaCluster.Status.NodePools == nil {
+		r.pandaCluster.Status.NodePools = map[string]vectorizedv1alpha1.NodePoolStatus{}
+	}
+
+	npStatus, ok := r.pandaCluster.Status.NodePools[r.nodePool.Name] // FIXME use NP name not STS name.
+	if !ok {
+		npStatus = vectorizedv1alpha1.NodePoolStatus{
+			CurrentReplicas: *r.nodePool.Replicas,
+		}
```

Comment:
question: why are setting the currentReplicas in the getStatus method? It smells to me on a first look as I don't fully understand the motivation behind.

Reply:
it needs to get initialized at some point. this way, i can control when this happens. every access to status goes through getStatus, and getStatus lazily initializes the nodePoolStatus if necessary.